packages/bcer-api/app/src/location/locationDataPortal.controller.ts

```diff
@@ -258,16 +258,15 @@ export class LocationDataPortalController {
   }
 
   @ApiOperation({ summary: 'Deletes a location by id' })
-  @ApiResponse({ status: HttpStatus.OK, type: LocationRO })
   @HttpCode(HttpStatus.OK)
   @UseGuards(AuthDataGuard)
   @Roles(ROLES.HA_ADMIN, ROLES.MOH_ADMIN)
   @AllowAnyRole()
   @Patch('/delete-location/:id')
   async deleteLocation(@Param('id') id: string){
     if(!id) throw NotFoundException;
-    const location = await this.service.hardDeleteLocation(id);
-    return location;
```

Comment:
Could we add in a check here to make sure the location we're trying to delete exists?

Reply:
We should also do a check here to make sure we can only delete a closed location, or one that has been deleted on the retailer side (soft delete). https://freshworks.atlassian.net/browse/VAPE-397. You can look up a location regardless of whether it's been soft deleted or not by including `{withDeleted: true}` in the query: https://github.com/typeorm/typeorm/blob/master/docs/find-options.md